ViewModel/ImpactAreaScenario/Editor/SpecificIASEditorVM.cs

```diff
@@ -435,15 +437,56 @@ public FdaValidationResult IsValid()
             vr.AddErrorMessage(IsRatingCurveValid());
             vr.AddErrorMessage(IsStageDamageValid());
 
-            //todo: actually run the compute and see if it was successful.
-
             if (!vr.IsValid)
             {
                 vr.InsertNewLineMessage(0, "Errors in Impact Area: " + CurrentImpactArea.Name);
             }
             return vr;
+        }
+
+        private void PreviewCompute()
+        {
+            AnalyticalFrequencyElement freqElem = SelectedFrequencyElement.ChildElement as AnalyticalFrequencyElement;
+            InflowOutflowElement inOutElem = SelectedInflowOutflowElement.ChildElement as InflowOutflowElement;
+            RatingCurveElement ratElem = SelectedRatingCurveElement.ChildElement as RatingCurveElement;
+            ExteriorInteriorElement extIntElem = SelectedExteriorInteriorElement.ChildElement as ExteriorInteriorElement;
+            LeveeFeatureElement leveeElem = SelectedLeveeFeatureElement.ChildElement as LeveeFeatureElement;
+            AggregatedStageDamageElement stageDamageElem = SelectedStageDamageElement.ChildElement as AggregatedStageDamageElement;
+
+            SimulationCreator sc = new SimulationCreator(freqElem, inOutElem, ratElem, extIntElem, leveeElem,
+                stageDamageElem, CurrentImpactArea.ID);
 
+            Simulation simulation = sc.BuildSimulation();
+
+            MeanRandomProvider mrp = new MeanRandomProvider();
+            try
+            {
+                metrics.Results result = simulation.Compute(mrp, 1);
+                Console.WriteLine("Mean ead: " + result.ExpectedAnnualDamageResults.MeanEAD("InteriorStageDamage"));
+                double ead = result.ExpectedAnnualDamageResults.MeanEAD("InteriorStageDamage");
+                double total = result.ExpectedAnnualDamageResults.MeanEAD("Total");
+                int i = 0;
+            }
+            catch (Exception e)
+            {
+                MessageBox.Show(e.Message);
+            }
         }
+
+        private List<UncertainPairedData> GetStageDamagesAsPairedData()
+        {
+            List<UncertainPairedData> stageDamages = new List<UncertainPairedData>();
+            List<StageDamageCurve> stageDamageCurves = GetStageDamageCurves();
+            foreach (StageDamageCurve curve in stageDamageCurves)
+            {
+                //todo: i don't like this.
```

Comment:
Why don't you like this?  or is this part of the call we're in?

Reply:
I changed the todo to be more descriptive of what i don't like. 
                //todo: i don't like that i have to take the icoordinates funtion and turn it into an ifda function to turn it into a paired data object.